File: educational-system-consumer-teacher-9002/src/main/java/edu/hubu/consumerteacher/service/ITeacherCourseService.java
package edu.hubu.consumerteacher.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.teacher.TeacherCourse;
import io.swagger.annotations.ApiImplicitParam;
import io.swagger.annotations.ApiImplicitParams;
import io.swagger.annotations.ApiOperation;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "teacher-provider-service")
public interface ITeacherCourseService {

    @GetMapping("/teacherCourse/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/teacherCourse/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "teacherCourseId") String orderBy);

    @GetMapping("/teacherCourse/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherCourseId") String orderBy
    );

    @PostMapping("/teacherCourse/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherCourseId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.TeacherCourse teacherCourse
    );

    @PostMapping("/teacherCourse/")
    CommonResult save(@RequestBody TeacherCourse teacherCourse);

    @PutMapping("/teacherCourse/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody TeacherCourse teacherCourse);


    @DeleteMapping("/teacherCourse/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/teacher/ITeacherCourseService.kt
package edu.hubu.commons.service.teacher

import edu.hubu.commons.model.teacher.TeacherCourse;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface ITeacherCourseService {
    fun save(t: TeacherCourse): TeacherCourse
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: TeacherCourse): TeacherCourse
    fun selectOneById(id: String): TeacherCourse
    fun selectAllOrderBy(order: String = "teacherCourseId"): List<TeacherCourse>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherCourseId"): List<TeacherCourse>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherCourseId", t: edu.hubu.commons.mybatis.model.TeacherCourse): List<edu.hubu.commons.mybatis.model.TeacherCourse>
}
